src/pybind/mgr/prometheus/module.py

```diff
@@ -793,8 +793,8 @@ def get_rbd_stats(self):
         pools_refreshed = False
         if pools:
             next_refresh = self.rbd_stats['pools_refresh_time'] + \
-                self.get_localized_module_option(
-                'rbd_stats_pools_refresh_interval', 300)
+                float(self.get_localized_module_option(
+                'rbd_stats_pools_refresh_interval', 300))
```

Comment:
@b-ranto are you sure this issue still happens in master, after #33991? 

Since #25597, ceph-mgr `_get_module_option()` returns a typed option (`PyUtil.cc`: `PyLong` for `int` and `PyFloat` for `float`). Perhaps the issue needs to be fixed by setting the MODULE_OPTION (and the default is also unnecessary here, as it's already pre-set)?:
```diff
- {'name': 'rbd_stats_pools_refresh_interval', 'type': 'int', 'default': 300},
+ {'name': 'rbd_stats_pools_refresh_interval', 'type': 'float', 'default': 300},
```

```suggestion
                self.get_localized_module_option(
                'rbd_stats_pools_refresh_interval')
```

Reply:
Hi Ernesto, not sure. We got a report downstream that this happens with nautilus and octopus. Octopus does seem to have the `type` fix https://github.com/ceph/ceph/pull/33991/commits/6d5f88450e61122016fcf7b6cf9431dc67128d3d in it.

@umangachapagain Can you elaborate a bit more on where you were able to hit this? Can you check that the version where you were able to hit this did already have a `type` and `default` set?

Reply:
I looked into the container and the fix mentioned by Ernesto was present (in ceph/ceph:v15.2.4). Couldn't test anything else as ceph commands weren't working due to some connection issue in my cluster.

Reply:
i concur with @epuertat on this. it's interesting why 6d5f88450e61122016fcf7b6cf9431dc67128d3d failed to populate the correct type. so before we have an answer to it, i'd prefer holding this change back, and merging it once we have at least a theory.